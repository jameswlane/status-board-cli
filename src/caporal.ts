import * as caporal from 'caporal';
import { readFileSync } from 'jsonfile';
import * as path from 'path';

const pkg = readFileSync(path.normalize(path.join(__dirname, '../package.json')));

caporal
  .version(pkg.version)
  .description(pkg.description)

  .command('example', 'hello world')
  .action((args, options, logger) => {
    logger.info('Command \'example\' called with:');
    logger.info('arguments: %j', args);
    logger.info('options: %j', options);
  });

export default caporal;
