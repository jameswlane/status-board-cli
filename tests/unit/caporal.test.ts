import caporal from '../../src/caporal';

jest.mock('jsonfile', () => {
  return {
    readFileSync: jest.fn(() => {
      return {
        description: 'Mocked description',
        version: '1.0.0',
      };
    }),
  };
});

describe('src/index.ts', () => {
  it('should match our snapshot', () => {
    expect(caporal).toMatchSnapshot();
  });
});
