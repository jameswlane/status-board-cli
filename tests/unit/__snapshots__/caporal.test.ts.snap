// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/index.ts should match our snapshot 1`] = `
Program {
  "_autocomplete": Autocomplete {
    "_completions": Object {},
    "_program": [Circular],
    "_tab": Complete {
      "_completions": Array [],
      "_events": Object {},
      "_eventsCount": 0,
      "_maxListeners": undefined,
      "_results": Array [],
      "domain": null,
      "options": Object {
        "cache": false,
        "name": "jest",
        "ttl": 300000,
      },
    },
  },
  "_bin": "jest",
  "_commands": Array [
    Command {
      "_action": [Function],
      "_alias": null,
      "_args": Array [],
      "_default": false,
      "_description": "hello world",
      "_lastAddedArgOrOpt": null,
      "_logger":  {
        "_events": Object {},
        "_eventsCount": 0,
        "_maxListeners": undefined,
        "_names": Array [
          "caporal",
        ],
        "debug": [Function],
        "domain": null,
        "emitErrs": false,
        "error": [Function],
        "exceptionHandlers": Object {},
        "exitOnError": true,
        "filters": Array [],
        "id": null,
        "info": [Function],
        "level": "info",
        "levels": Object {
          "debug": 4,
          "error": 0,
          "info": 2,
          "silly": 5,
          "verbose": 3,
          "warn": 1,
        },
        "padLevels": false,
        "profilers": Object {},
        "rewriters": Array [],
        "silly": [Function],
        "stripColors": false,
        "transports": Object {
          "caporal": CaporalTransport {
            "_onError": [Function],
            "level": "info",
            "name": "caporal",
          },
        },
        "verbose": [Function],
        "warn": [Function],
      },
      "_name": "example",
      "_options": Array [],
      "_program": [Circular],
      "debug": [Function],
      "description": [Function],
      "error": [Function],
      "info": [Function],
      "log": [Function],
      "warn": [Function],
    },
  ],
  "_defaultCommand": null,
  "_description": "Mocked description",
  "_helper": Help {
    "_customHelp": Object {},
    "_program": [Circular],
  },
  "_logger":  {
    "_events": Object {},
    "_eventsCount": 0,
    "_maxListeners": undefined,
    "_names": Array [
      "caporal",
    ],
    "debug": [Function],
    "domain": null,
    "emitErrs": false,
    "error": [Function],
    "exceptionHandlers": Object {},
    "exitOnError": true,
    "filters": Array [],
    "id": null,
    "info": [Function],
    "level": "info",
    "levels": Object {
      "debug": 4,
      "error": 0,
      "info": 2,
      "silly": 5,
      "verbose": 3,
      "warn": 1,
    },
    "padLevels": false,
    "profilers": Object {},
    "rewriters": Array [],
    "silly": [Function],
    "stripColors": false,
    "transports": Object {
      "caporal": CaporalTransport {
        "_onError": [Function],
        "level": "info",
        "name": "caporal",
      },
    },
    "verbose": [Function],
    "warn": [Function],
  },
  "_supportedShell": Array [
    "bash",
    "zsh",
    "fish",
  ],
  "_version": "1.0.0",
  "bin": [Function],
  "description": [Function],
  "logger": [Function],
  "name": [Function],
  "version": [Function],
}
`;
